### PR TITLE
Add uniquifying values to Esti tests with external storage access

### PR DESCRIPTION
**Pull request by arielshaqed**
* Add a random suffix to _all_ external `s3://...` URLs used in Esti.
 * Split metadata client test into separate GC, export modes.  This allows them to run in parallel and makes Esti succeed more often.

   Not really _required_ for this PR... but it **does** make Esti pass rather than time out when GitHub actions are loaded, so _helpful_ to produce this PR.

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]
lgtm

File: clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala [hunk 1]
```diff
@@ -366,7 +366,7 @@ object GarbageCollector {
 
   private def repartitionBySize(df: DataFrame, maxSize: Int, column: String): DataFrame = {
     val nRows = df.count()
-    val nPartitions = math.max(1, math.floor(nRows / maxSize)).toInt
+    val nPartitions = math.max(1, math.ceil(nRows / maxSize)).toInt
```
Comment (johnnyaug) [inline clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala:369]: Why part of this PR?

File: .github/workflows/esti.yaml [hunk 2]
```diff
@@ -613,37 +637,88 @@ jobs:
...
         working-directory: test/spark
         run: ./run-gc-test.sh
 
+      - name: lakeFS Logs on Spark with gateway failure
+        if: ${{ failure() }}
+        continue-on-error: true
+        working-directory: test/spark
+        run: docker-compose logs --tail=1000 lakefs
+
+  metadata-client-export-spark3:
```
Comment (johnnyaug) [inline .github/workflows/esti.yaml:661]: New test? Is this related to this PR?

**Review by johnnyaug — CHANGES_REQUESTED** [review]
Some parts here look like they belong to a different PR.
Can you go over the changed files?
If they are relevant, can you change the PR description to reflect it?

File: .github/workflows/esti.yaml (hunk 2 above)
Comment (arielshaqed) [reply .github/workflows/esti.yaml:661]: Sort of, but mostly in order to pass tests.  In eb8df2b19048ef8173b6e1874b79db30b1e0331b I split the metadata client test into gc and export so that they may run in parallel -- that lets the export test (just a few minutes) run without timing out, and also helps the long-running gc test finish running on time.
Without this, Esti barely managed to sneak in under the deadline -- which would have meant we cannot trust this PR.
I shall update the PR description to reflect this addition. :-)  So now we won't have a dirty hack, we will have a **documented** dirty hack.

File: clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala (hunk 1 above)
Comment (arielshaqed) [reply clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala]: Moved it out.  (It's a bug, but not one we need to fix here.)

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL, especially at the updated description.

**Review by lynnro314 — APPROVED** [review]

**Review by johnnyaug — APPROVED** [review]

**Comment by arielshaqed** [comment]
THANKS!

Let's hope now everything works :-) ❗   Except that Esti is still too slow, of course.
